[Benjy Tan] iP

benjytan45678 opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
#### DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use
#### All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smiley:

#### And it is **FREE!**
#### Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

#### If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

benjytan45678 committed:
Level-1

benjytan45678 committed:
Level-2

benjytan45678 committed:
Level-3

benjytan45678 committed:
Level-4

benjytan45678 committed:
A-TextUiTesting

benjytan45678 committed:
Level-5

benjytan45678 committed:
Level-6

benjytan45678 committed:
Updated changes, with more exceptions

benjytan45678 committed:
Added Level-7

benjytan45678 committed:
Add Level-8

benjytan45678 committed:
Refactor Level-7 and 8

benjytan45678 committed:
Add loading file functionality

benjytan45678 committed:
Add MoreOOP

benjytan45678 committed:
Refactor classes into packages

benjytan45678 committed:
Merge remote-tracking branch 'fork/add-gradle-support'

benjytan45678 committed:
Add JUnit tests

benjytan45678 committed:
Add JavaDoc

benjytan45678 committed:
Add coding standard

benjytan45678 committed:
Add Level 9

benjytan45678 committed:
Merge branch-A-JavaDoc with master

benjytan45678 committed:
Merge branch-A-CodingStandard with master

benjytan45678 committed:
Merge branch-Level-9 with master

benjytan45678 committed:
Add date-time recognition to Event class

benjytan45678 committed:
Add Exception to mark, unmark and delete command class

nseah21 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,109 @@
+                DefaultCommand defaultCommand = new DefaultCommand();
+                return defaultCommand;
+
+        }
```
I like how you followed the recommended coding standard for switch statements 👍

nseah21 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,21 @@
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+import java.util.ArrayList;
```
Should this import be re-ordered?

nseah21 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,64 @@
+
+
+    /**
+     * Return completion status of the task
```
Perhaps the Javadoc should start with returns instead of return 🤔

nseah21 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,33 @@
+     * @param description The error description for the exception.
+     */
+    public DukeException(String description) {
+
```
Should this blank line be removed?

nseah21 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,135 @@
+     */
+    public void showLine() {
+
+        System.out.println("____________________________________________________________\n");
```
Should the indentation for this line be modified?

nseah21 reviewed (commented):
Good work! Some unexpected blank lines but they are not problematic. 💯

benjytan45678 committed:
Add GUI to Duke

benjytan45678 committed:
Add checkstyle

benjytan45678 committed:
Merge branch-Level-10 to master

benjytan45678 committed:
Add Color to GUI

benjytan45678 committed:
AddCommand and DeleteCommand assumes number of tasks in taskList is
mutated correctly while the Parser assumes keyword for the FindCommand
does not start with whitespace.

Make sure these assumptions hold true

Use assertions to check for violation in assumptions and print out
relevant alert messages

Ensure programme runs as expected

benjytan45678 committed:
Parser parses the dateTime for deadline and event separately, even though they are similar

Make code more readable and improve code quality

Use abstraction for the methods to deal with localTime and localDate used for deadline and event

Method names are shorter and easier to understand

benjytan45678 committed:
Merge pull request #1 from benjytan45678/branch-A-Assertions

Branch A-Assertions

benjytan45678 committed:
Merge pull request #2 from benjytan45678/branch-A-CodeQuality

Branch A-CodeQuality

benjytan45678 committed:
Add D-Contacts

benjytan45678 committed:
Add user guide and fix bugs

benjytan45678 committed:
Merge branch-D-Contacts-Extension to master
